### PR TITLE
Delta Lake Diff MVP: Add Delta Lake differ

### LINKED ISSUE #4966: Delta Diff MVP - Task: Business Logic
Add the business logic to the diff service: according to the received type (delta), use the client abstraction to call the corresponding plugin, receive the answer and compute a response.
- Create a Delta Diff gRPC client.

- [x] Lean version
- [ ] Fully functional
- [ ] Tests

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
### Linked Issue

Closes #4966

---

## Change Description

- Add a Delta Lake differ and plugin implementation.
- Extracted the plugin properties outside of the internal package so that they could be configured...

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]

File: pkg/plugins/diff/delta_differ.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		}
+		return Response{}, err
+	}
+	return Response{
+		Diffs:      buildDiffEntries(dr),
+		ChangeType: ChangeType(dr.ChangeType),
+	}, nil
+}
+
+func buildDiffEntries(dr *DiffResponse) []DiffEntry {
```
Comment (talSofer) [inline pkg/plugins/diff/delta_differ.go:58]: nit; should we decide between record/entry and go with one term?

**Review by talSofer — COMMENTED** [review]
Looks good, what is the plan for testing delta lake differ?

File: pkg/plugins/diff/delta_differ.go (hunk 1 above)
Comment (Jonathan-Rosenberg) [reply pkg/plugins/diff/delta_differ.go:58]: decided 😄

**Comment by talSofer** [comment]
> @talSofer, regarding:
> 
> > what is the plan for testing delta lake differ?
> 
> I don't see a lot of value in unit testing this code as it has a very basic contract. Would you feel more confident if I'll add this test?

@Jonathan-Rosenberg adding a test will make sure that we don't break the contract even if it is simple (i.e. the response structure etc), so I think it is worth adding it.

File: pkg/plugins/diff/delta_differ_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		{
+			name:   "success",
+			client: TableDifferClientMock{},
+			diffResponse: &DiffResponse{
+				Entries: []*TableOperation{
+					{
+						Id:        "id1",
+						Timestamp: timestamppb.Now(),
+						Operation: "OPA",
+						Content: map[string]string{
```
Comment (talSofer) [inline pkg/plugins/diff/delta_differ_test.go:35]: I guess that an updated version will include operation type?

**Review by talSofer — APPROVED** [review]
Looks good, thanks for adding tests! 
please look into adding the operation type into the diff entry

File: pkg/plugins/diff/delta_differ_test.go (hunk 2 above)
Comment (Jonathan-Rosenberg) [reply pkg/plugins/diff/delta_differ_test.go:35]: good catch

**Pull request merged**